frontend/src/lib/DeltaParser.ts

```diff
@@ -54,19 +60,50 @@ export function applyDelta(
 
   dispatchOneOf(delta, "type", {
     newElement: (element: SimpleElement) => {
```

Comment:
I think you can just use `mergeIn` instead of `setIn` like this:

```js
      elements[container] = elements[container].mergeIn(
        deltaPath,
        handleNewElementMessage(container, element, reportId, metadata)
      )
```
...and update `handleNewElementMessage` to return an `elementWrapper`.

This way `elements[container]` will be a new `elementWrapper` but the `element` inside the wrapper will only change if the proto changed.

Reply:
We have two different problems if we want to use mergeIn, the first time elements comes with a diferrent structure than the needed one, so that's why we were using setIn always in a past, but now as we need to mutation just certains options i have to do an update and a withMutation()... 
I don't know if it's possible to do something like that with merge and regenerating the first time the structure

Reply:
Makes sense. Good catch!

But I think the logic in the PR could be much simpler, like this:

```ts
newElement: (element: SimpleElement) => {
  let elementWrapper: ElementWrapper = elements[container].getIn(deltaPath)

  const element = handleNewElementMessage(container, element, reportId, metadata)
  elementWrapper = elementWrapper.update("element", element)

  elementWrapper = elementWrapper.set("reportId", reportId)
  elementWrapper = elementWrapper.set("metadata", metadata)

  elements[container] = elements[container].setIn(deltaPath, elementWrapper)
},
```

Reply:
That won't work because you are using a setIn and will create a new instance, and also you are trying to update the element property which could be undefined

Reply:
Wait, unless I'm misunderstanding something, we definitely _want_ to create a new instance since we want to update the `reportId` every time.

Reply:
newElement: (element: SimpleElement) => {
  let elementWrapper: ElementWrapper = elements[container].getIn(deltaPath)

  const element = handleNewElementMessage(container, element, reportId, metadata)
  elementWrapper = elementWrapper.update("element", (oldElement) => {
    return oldElement.equals(element) ? oldElement : element
  })

  elementWrapper = elementWrapper.set("reportId", reportId)
  elementWrapper = elementWrapper.set("metadata", metadata)

  elements[container] = elements[container].setIn(deltaPath, elementWrapper)
},

Reply:
And actually, if we pipe in the `ForwardMsg.hash` property over here, we can avoid doing the somewhat expensive `.equals()` check.